### PR TITLE
reimplement flows in pyro

### PR DESCRIPTION
**Pull request by deepchatterjeeligo**

### CODE REVIEW

**Comment by alecgunny** [comment]
Just took a quick look at the code, and out of curiosity is there a reason to not have the `NormalizingFlow` class inherit from `torch.nn.Module`, that way you can inherit all of the `parameters` and `state_dict` functionality?

**Comment by deepchatterjeeligo** [comment]
> is there a reason to not have the NormalizingFlow class inherit from torch.nn.Module

Nope. That should have been done right from the start.
